timesheets/Code.gs

```diff
@@ -0,0 +1,144 @@
+        var message = 'NOT APPROVED';
+        var subject = 'TimeSheet not Approved';
+      } else if (dropdownValue.getValue() == 'IN PROGRESS') {
+        continue;
+      } 
+      
+      MailApp.sendEmail(email, subject, message);
+      var notifiedUpdate = sheet.getRange(i, lastCol).setValue('NOTIFIED');
+    }
+}
+
+
+/**
+* Runs the "Add Approval Column" script, adding an approval column at the end of the sheet, 
+* made entirely of checkboxes.
+* Adds "NOTIFIED STATUS" column indicating whether or not an employee has yet been e mailed.
+*/
+function addApprovalColumn() { 
+  // adds new column
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastCol = sheet.getLastColumn();
+  var lastRow = sheet.getLastRow();
+  sheet.insertColumnAfter(lastCol);
+  
+  // sets new column title
+  var approvalColumn = sheet.getRange(1, lastCol + 1).setBackground('ffe3ee').setValue('APPROVAL');
+  var FROZEN_ROWS = sheet.getFrozenRows();
+
+  // make sure approval column is all drop-down menu
+  for (var i = FROZEN_ROWS + 1; i <= lastRow; i++) {
+    var dropdownBox = sheet.getRange(i,lastCol + 1);
+    var dropdownValues = ['APPROVED', 'NOT APPROVED', 'IN PROGRESS'];
+    var rule = SpreadsheetApp.newDataValidation().requireValueInList(dropdownValues).build();
+    dropdownBox.setDataValidation(rule);
+    dropdownBox.setValue('IN PROGRESS');
+  }
+  
+  // add the notified column
+  sheet.insertColumnAfter(APPROVAL_COLUMN); //global
+  var notifiedColumn = sheet.getRange(1, APPROVAL_COLUMN + 1).setBackground('ffe3ee').setValue('NOTIFIED STATUS'); 
+  
+  // sets notified status to NOT NOTIFIED
+  for (var i = FROZEN_ROWS + 1; i <= lastRow; i++) {
+    var notifiedCell = sheet.getRange(i, APPROVAL_COLUMN + 1);
+    var notifiedValues = ['NOTIFIED', 'NOT NOTIFIED'];
+    var rule = SpreadsheetApp.newDataValidation().requireValueInList(notifiedValues).build();
+    notifiedCell.setDataValidation(rule);
+    notifiedCell.setValue('NOT NOTIFIED');
+  }
+}
+
+/**
+* Runs the "Calculate Pay" script, creating a new column with each employee's 
+* pay based on their reported hours and hourly wage.
+*/
+function calculatePay() {
+  // adds new column
+  var sheet = SpreadsheetApp.getActiveSheet();
+  var lastCol = sheet.getLastColumn();
+  var lastRow = sheet.getLastRow();
+  sheet.insertColumnAfter(lastCol);
+  
+  // sets new column title
+  var weeklyPayColumn = sheet.getRange(1, lastCol + 1).setValue('WEEKLY PAY');
+  var rangeValues = sheet.getRange(FROZEN_ROWS + 1, HOURS_START, 6, lastCol).getValues();
+  var FROZEN_ROWS = sheet.getFrozenRows();
+  
+  // goes through every employee
+  for (var i = 0; i <= lastRow - (FROZEN_ROWS + 1); i++) {
+    var numHours = 0;
+    for (var j = 0; j < 5; j++) {
+      numHours += rangeValues[i][j];
+    }
+
+    // calulates pay
+    var weeklyPay = numHours * rangeValues[i][HOURLY_PAY_COLUMN - HOURS_START];
+    var setPay = sheet.getRange(i + (FROZEN_ROWS + 1), lastCol + 1).setValue(weeklyPay);
```

Comment:
You're still setting the pay one row at a time, instead of the whole column at a time. You want to build up a 2D array of values and then set them all at once with `setValues()`.